snuba/query/processors/granularity_processor.py

```diff
@@ -0,0 +1,35 @@
+from snuba.datasets.metrics import DEFAULT_GRANULARITY
+from snuba.query.conditions import ConditionFunctions, binary_condition
+from snuba.query.expressions import Column, Literal
+from snuba.query.logical import Query
+from snuba.query.processors import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+
+#: Granularities for which a materialized view exist, in ascending order
+GRANULARITIES_AVAILABLE = (10, 60)
+
+
+class GranularityProcessor(QueryProcessor):
+    """ Use the granularity set on the query to filter on the granularity column """
+
+    @staticmethod
+    def __get_granularity(query: Query) -> int:
+        """ Find the best fitting granularity for this query """
+        requested_granularity = query.get_granularity()
+        if requested_granularity:
+            for granularity in reversed(GRANULARITIES_AVAILABLE):
+                if (requested_granularity % granularity) == 0:
+
+                    return granularity
+
+        return DEFAULT_GRANULARITY
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        granularity = self.__get_granularity(query)
```

Comment:
Shouldn't we reject queries that have a granularity that is not a multiple of 10 seconds or a multiple of one of the granularities we stored.

Please create a subclass of https://github.com/getsentry/snuba/blob/master/snuba/query/exceptions.py#L5 for that.

Actually, better, create an entity validator for that 
https://github.com/getsentry/snuba/blob/master/snuba/datasets/entity.py#L103-L110